components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -0,0 +1,228 @@
+
+        ClaimMapping[] coreClaims = new ClaimMapping[]{cMap1, cMap2, cMap3, cMap4};
+
+        HashMap<String, Boolean> requiredAttributes = new HashMap<String, Boolean>() {
+            {
+                put("test1.test", true);
+            }
+        };
+
+        String[] roles = new String[]{"role1", "role2", "role3"};
+
+        return new Object[][]{
+                {coreClaims, requiredAttributes, roles}
+        };
+    }
+
+    @Test(dataProvider = "ClaimData")
+    public void testGetMe(Object[] cMap, HashMap<String, Boolean> required, String[] userRoles) throws Exception {
+
+        when(mockedClaimManager.getAllClaimMappings(anyString())).thenReturn((ClaimMapping[]) cMap);
+
+        SCIMUserSchemaExtensionBuilder sb = spy(new SCIMUserSchemaExtensionBuilder());
+        mockStatic(SCIMUserSchemaExtensionBuilder.class);
+        when(SCIMUserSchemaExtensionBuilder.getInstance()).thenReturn(sb);
+        when(sb.getExtensionSchema()).thenReturn(mockedScimAttributeSchema);
+
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn("testPrimaryDomain");
+
+        when(mockedUserStoreManager.getSecondaryUserStoreManager(anyString())).thenReturn(mockedUserStoreManager);
+        when(mockedUserStoreManager.isSCIMEnabled()).thenReturn(true);
+        when(mockedUserStoreManager.getRoleListOfUser(anyString())).thenReturn(userRoles);
+        mockStatic(AttributeMapper.class);
+        when(AttributeMapper.constructSCIMObjectFromAttributes(anyMap(), anyInt())).thenReturn(mockedUser);
+        when(mockedUserStoreManager.getRealmConfiguration()).thenReturn(mockedRealmConfig);
+        when(mockedRealmConfig.getEveryOneRoleName()).thenReturn("roleName");
+        when(mockedUserStoreManager.getTenantId()).thenReturn(1234567);
+        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
+
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager, mockedClaimManager);
+        assertNotNull(scimUserManager.getMe("testUserName", required));
+    }
+
+    @Test(dataProvider = "groupName")
+    public void testGetGroup(Object roleName, String userStoreDomain, Object expected) throws Exception {
+
+        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
+        when(mockedGroupDAO.getGroupNameById(anyInt(), anyString())).thenReturn((String) roleName);
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn(userStoreDomain);
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager, mockedClaimManager);
+        Group result = scimUserManager.getGroup("1234567", new HashMap<String, Boolean>());
+
+        String actual = null;
+        if (result != null) {
+            actual = result.getDisplayName();
+        }
+        assertEquals(actual, expected);
```

Comment:
With unit tests, most of the 'if' conditions can be changed to assert statements. How about the following code instead of above code lines 182-186?

```
assertNotNull(result);
assertEquals(result.getDisplayName(), expected);
```

Reply:
There's a scenario where the asserted string should be null so the approach you have suggested is not applicable here. So shall I keep it this way or is there any better way to do it?

Reply:
Then as I understand, the result.getDisplayName() should be null, not the result itself, isn't it?